app/assets/src/components/ui/controls/GeoSearchInputBox.jsx

```diff
@@ -1,11 +1,43 @@
 import React from "react";
 import PropTypes from "prop-types";
-import { isString } from "lodash/fp";
+import { compact, get, isString } from "lodash/fp";
 
 import { logAnalyticsEvent } from "~/api/analytics";
 import { getGeoSearchSuggestions } from "~/api/locations";
 import LiveSearchPopBox from "~ui/controls/LiveSearchPopBox";
 
+export const LOCATION_PRIVACY_WARNING =
+  "Changed to county/district level for personal privacy.";
+export const LOCATION_UNRESOLVED_WARNING =
+  "Unresolved plain text, not shown on maps.";
+
+// For human samples, drop the city part of the name and show a warning.
```

Comment:
Shouldn't this be enforced on the server? Clients are vulnerable.
Remind me again, but we still narrows it down when saving, right?

Reply:
Yes and yes. I clarified the comment to emphasize that.